homeassistant/components/recorder/migration.py

```diff
@@ -41,20 +41,39 @@ def _apply_update(engine, new_version):
     from sqlalchemy import Table
     from . import models
 
-    if new_version == 1:
-        def create_index(table_name, column_name):
-            """Create an index for the specified table and column."""
-            table = Table(table_name, models.Base.metadata)
-            name = "_".join(("ix", table_name, column_name))
-            # Look up the index object that was created from the models
-            index = next(idx for idx in table.indexes if idx.name == name)
-            _LOGGER.debug("Creating index for table %s column %s",
-                          table_name, column_name)
-            index.create(engine)
-            _LOGGER.debug("Index creation done for table %s column %s",
-                          table_name, column_name)
+    def create_index(table_name, column_name):
+        """Create an index for the specified table and column."""
+        table = Table(table_name, models.Base.metadata)
+        name = "_".join(("ix", table_name, column_name))
+        # Look up the index object that was created from the models
+        index = next(idx for idx in table.indexes if idx.name == name)
+        _LOGGER.debug("Creating index for table %s column %s",
+                      table_name, column_name)
+        index.create(engine)
+        _LOGGER.debug("Index creation done for table %s column %s",
+                      table_name, column_name)
+
+    def create_compound_index(table_name, column_names):
+        """Create an index for the specified table and columns."""
+        table = Table(table_name, models.Base.metadata)
+        index_name = "ix_" + table_name + "_" + "_".join(column_names)
```

Comment:
:bulb: 

Aha ok that helps explain things quite a bit! If it's ok, I'll probably wrap the above explanation around some in-line comments since that might be helpful to future folks trying to figure out how that works.